apis/CommentsEndpoint.es6.js

```diff
@@ -53,16 +53,14 @@ const parseGetBody = (apiResponse, hasChildren) => {
       });
     }
 
-    comments = parseCommentList(body[1].data.children);
+    comments = body[1].data.children.map(parseCommentData);
   } else if (body.json && body.json.data) {
-    if (hasChildren) { // treeify 'load more comments' replies
-      comments = treeifyComments(parseCommentList(body.json.data.things));
-    } else {
-      comments = parseCommentList(body.json.data.things);
-    }
+
+    const { things } = body.json.data;
+    comments = treeifyComments(things.map(parseCommentData));
   }
 
-  normalizeCommentReplies(comments, (commentJSON, isTopLevel) => {
+  normalizeCommentReplies(comments, true, (commentJSON, isTopLevel) => {
```

Comment:
So one thing we can do is just to used named arguments, which is something I had but removed for some reason, probably line length issues. I can add it back in to get the same effect. Where possible, I generally like to avoid adding function layers. It makes reading the code just slightly more difficult.